daisy_workflows/image_build/debian/build.py

```diff
@@ -123,6 +124,50 @@ def main():
   synopsis_dest = os.path.join(outs_path, 'synopsis.json')
   utils.UploadFile('/tmp/synopsis.json', synopsis_dest)
 
+  # Create and upload metadata of the image and packages
+  logging.info('Creating image metadata.')
+  image = {
+      "family": "debian-9",
+      "name": re.match("debian-9-[\\w-]*", image_dest),
+      "version": "stretch",
+      "location": image_dest,
+      "release_date": datetime.now(timezone.utc),
+      "packages": []
+  }
+  # Read list of guest package
+  with open("guest_package") as f:
+    guest_packages = f.read().splitlines()
+
+  for package in guest_packages:
+    cmd = "dpkg-query -W --showformat '${Package}:${Version}:${Git}'" \
+          + package
+    code, stdout = utils.Excute(cmd, capture_output=True)
+    if code == 0:
+      splits = stdout.decode('utf-8').split(':')
+      package_name = splits[0]
+      package_version = splits[1]
+      package_git_url = splits[2]
+      start = package_version.index(":")
+      end = package_version.rindex(".")
+      package_release_date = package_version[start: end]
```

Comment:
uh given you did split on ':' already, you shouldn't have any more to index off of here. if you did, it looks like you're trying to pull a date out of a version, and i think you should just store the version as-is and not try to parse it. also the version contains the build date, not the release date.

Reply:
I change the field "release_date" to "build_date" for both image and package. We store the version as-is. and I want to include a "build_date" field which the type in spanner is Date for query. But to me, the "version" is the only "source" we can get this. At least, we need it to parse/convert from it somewhere.

Reply:
i should have made this two comments.

comment 1, code error: the `.index(":")` should not work since you have already called `.split(':')` on this string. 

comment 2, we should not parse "1:20200707.01.g1" into "20200707". Just pull the entire version field and don't bother about parsing it. In fact, you should append the release onto the version.

Reply:
I have fixed this. I  put the full version string.